include/astshim/CmpFrame.h

```diff
@@ -74,7 +74,9 @@ class CmpFrame : public Frame {
     explicit CmpFrame(Frame const &frame1, Frame const &frame2, std::string const &options = "")
             : Frame(reinterpret_cast<AstFrame *>(astCmpFrame(const_cast<AstObject *>(frame1.getRawPtr()),
                                                              const_cast<AstObject *>(frame2.getRawPtr()),
-                                                             "%s", options.c_str()))) {}
+                                                             "%s", options.c_str()))) {
+        assertOK();
+    }
```

Comment:
This is redundant because the `Frame` constructor calls the `Mapping` constructor which calls `assertOK`, and after that is called no AST code is executed that can set the flag. I admit that `astIsA<x>` is often called in these constructors, but:
* I don't think that can set the OK flag false
* Even if it can, this would only happen if it returned false, and at that point it's too late to call assertOK as an exception is already being raised

Where was the missing assertOK that triggered this bug? I think almost all the ones you added are redundant.

Reply:
I have not followed the code paths, but I put `assertOK` calls after any `ast*` calls (except for `astIsA*`). I figure it's better to be more careful than less careful, because it's a cheap test and if anything is missed, it manifests as a completely weird bug, and takes a full day of mucking around with `gdb` to understand it.

Reply:
That's a good answer. OK. Go ahead.